com.unity.multiplayer.mlapi/Prototyping/NetworkedAnimator.cs

```diff
@@ -14,447 +12,297 @@ namespace MLAPI.Prototyping
+        private void SendTrackedParams()
         {
-            if (EnableProximity)
+            var animParams = GetAnimParams( /* trackedOnly = */ true);
+
+            if (IsServer)
             {
-                List<ulong> clientsInProximity = new List<ulong>();
-                foreach (KeyValuePair<ulong, NetworkedClient> client in NetworkingManager.Singleton.ConnectedClients)
+                var clientRpcParams = new ClientRpcParams
                 {
-                    if (client.Value.PlayerObject == null || Vector3.Distance(transform.position, client.Value.PlayerObject.transform.position) <= ProximityRange)
-                        clientsInProximity.Add(client.Key);
-                }
-                InvokeClientRpcPerformance(ApplyAnimParamMsg, clientsInProximity, stream);
+                    Send = new ClientRpcSendParams
+                    {
+                        TargetClientIds = NetworkingManager.Singleton.ConnectedClientsList
+                            .Where(c => c.ClientId != NetworkingManager.Singleton.ServerClientId)
+                            .Select(c => c.ClientId)
+                            .ToArray()
+                    }
+                };
+                UpdateTrackedParamsClientRpc(animParams, clientRpcParams);
             }
             else
             {
-                InvokeClientRpcOnEveryoneExceptPerformance(ApplyAnimParamMsg, OwnerClientId, stream);
+                UpdateTrackedParamsServerRpc(animParams);
             }
         }
 
-        [ClientRpc]
-        private void ApplyAnimParamMsg(ulong clientId, Stream stream)
+        private void SendAllParamsAndState(int animStateHash, float animStateTime)
         {
-            ReadParameters(stream, true);
-        }
+            var animParams = GetAnimParams();
 
-        [ServerRpc]
-        private void SubmitAnimTriggerMsg(ulong clientId, Stream stream)
-        {
-            if (EnableProximity)
+            if (IsServer)
             {
-                List<ulong> clientsInProximity = new List<ulong>();
-                foreach (KeyValuePair<ulong, NetworkedClient> client in NetworkingManager.Singleton.ConnectedClients)
+                var clientRpcParams = new ClientRpcParams
                 {
-                    if (client.Value.PlayerObject == null || Vector3.Distance(transform.position, client.Value.PlayerObject.transform.position) <= ProximityRange)
-                        clientsInProximity.Add(client.Key);
-                }
-                InvokeClientRpcPerformance(ApplyAnimTriggerMsg, clientsInProximity, stream);
+                    Send = new ClientRpcSendParams
+                    {
+                        TargetClientIds = NetworkingManager.Singleton.ConnectedClientsList
+                            .Where(c => c.ClientId != NetworkingManager.Singleton.ServerClientId)
+                            .Select(c => c.ClientId)
+                            .ToArray()
```

Comment:
I'm not a big fan of LINQ method chains but left micro-optimization for later here.